torch/csrc/autograd/python_engine.cpp

```diff
@@ -57,6 +57,11 @@ void PythonEngine::thread_init(int device, const std::shared_ptr<ReadyQueue>& re
   pybind11::gil_scoped_acquire gil;
   pybind11::gil_scoped_release no_gil;
   Engine::thread_init(device, ready_queue, false);
+
+  if (should_increment) {
```

Comment:
Was it a bug that this was missing?

Reply:
Well it wasn't really a bug since it kinda worked. PyThonEngine would increment the count and pass false to Engine. Then, engine would always decrement without checking `should_increment`. I just felt it was kinda weird how both the classes had this contract with each other :) This just makes each one consistent on its own.

Reply:
Haha, thanks for the explanation.
We do have a growing user-base for libtorch that uses the non-python version so definitely a good fix !